### PR TITLE
Clean up the Ability Score builder, again and by a different person

### PR DESCRIPTION
**Pull request by fryguy1013**
Several things are broken by the changes that have been made to "improve" it.

Also, improve behavior of key ability score being set manually.

### CODE REVIEW

File: static/templates/actors/ability-builder.html [hunk 1]
```diff
@@ -135,7 +135,7 @@ <h2>{{localize "PF2E.Actor.Character.AbilityBuilder.Flaws"}}</h2>
                 <div class="full-row">{{localize "PF2E.Actor.Character.AbilityBuilder.BackgroundMissingHelp"}}</div>
             {{/if}}
 
-            <div class="hover-content" class="background-tooltip">
+            <div class="hover-content" id="background-tooltip">
```
Comment (CarlosFdez) [inline static/templates/actors/ability-builder.html:138]: The value of id must be unique within the entire page. This shouldn't use it.

Comment (CarlosFdez) [reply static/templates/actors/ability-builder.html:138]: That said, it seems that data-tooltip-content seems to use ids everywhere, even in the parts that should be unique. I guess without using tooltipster document side, it has to use ids unfortunately.

Might be better to use .tooltipster() for this like most of the codebase uses?

Comment (fryguy1013) [reply static/templates/actors/ability-builder.html]: It is using tooltipster like most of the rest of the codebase uses.

Comment (CarlosFdez) [reply static/templates/actors/ability-builder.html]: I meant feeding in the content directly via the ability builder application. But making the ids unique is fine enough.

File: src/module/actor/character/index.ts [hunk 2]
```diff
@@ -102,6 +102,9 @@ class CharacterPF2e extends CreaturePF2e {
     class!: Embedded<ClassPF2e> | null;
     deity!: Embedded<DeityPF2e> | null;
 
+    // base ability scores, before any rule element breaks things
+    baseAbilityScores!: Abilities | null;
+
```
Comment (stwlam) [inline src/module/actor/character/index.ts:107]: This is unnecessary and can get acquired in the builder application

File: src/module/actor/sheet/popups/ability-builder.ts [hunk 3]
```diff
@@ -3,14 +3,15 @@ import { Abilities } from "@actor/creature/data";
 import { AbilityString } from "@actor/types";
 import { ABILITY_ABBREVIATIONS } from "@actor/values";
 import { AncestryPF2e, BackgroundPF2e, ClassPF2e } from "@item";
+import { ActorPF2e } from "@actor/index";
 
-export class AbilityBuilderPopup extends Application {
+export class AbilityBuilderPopup extends FormApplication<ActorPF2e> {
```
Comment (stwlam) [inline src/module/actor/sheet/popups/ability-builder.ts:8]: There's no point in making this a `FormApplication` only to later prevent it from being submitted.

File: src/module/actor/character/index.ts (hunk 2 above)
Comment (fryguy1013) [reply src/module/actor/character/index.ts:107]: There is no place that the after calculated ability scores are available. This is for builder mode too.

Comment (stwlam) [reply src/module/actor/character/index.ts:107]: How about the bases be set for each score in the abilities object?

Comment (fryguy1013) [reply src/module/actor/character/index.ts:107]: ok i'll do that

**Pull request merged**